test/extensions/filters/network/thrift_proxy/decoder_test.cc

```diff
@@ -448,6 +470,31 @@ TEST_P(DecoderStateMachineValueTest, MapValueValue) {
   EXPECT_EQ(dsm.currentState(), ProtocolState::MapEnd);
 }
 
+TEST_P(DecoderStateMachineValueTest, IncompleteMapValues) {
```

Comment:
We need an Incomplete MapKeys test that performs this test again the map key (and matching code in the decoder to handle this case).